### ISSUE TITLE
Page/scroll gets error if children are updated

### ISSUE THREAD

**Issue by aarowman** [sag1v/react-elastic-carousel#39]
**Describe the bug**
If you update the children of the carousel after scrolling, then the active page is retained and can cause errors.

**To Reproduce**
1. Set children to 10 items
2. Click next until on the last page
3. Dynamically update children to 5 items (anything less than 9)
4. Click PREV arrow
5. See error: TypeError: child is undefined

Note: if using more than one element per page, then the current page may still work but show fewer elements, 
e.g. 
1. Set children to 10 items, page size 5
2. Click next (now on the last page, elements 5-10, [indices 4-9])
3. Dynamically update children to 7 items
4. See only last 2 items (since it was already scrolled past the first 5)

**Expected behavior**
First choice - updating children would automatically go to first page. (Current workaround is to set page in a parent component using ref.goTo() function when updating children)

Second choice: changing pages has a max index. (e.g. if index >= children list size, then set to grab highest index)

**Screenshots**
React viewer shows error: 
**TypeError: child is undefined**
```javascript 
convertChildToCbObj = index => {
  const { children } = this.props;
  const child = children[index]
  return { item: child.props, index };
};
```
(error at line 4: child.props)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Firefox
 - Version: 60.9.0esr (32-bit)

**Smartphone (please complete the following information):**
N/A

**Additional context**
N/A

**Comment by sag1v** [comment]
Great catch!
So in other words, the carousel is not supporting reducing the number of items just adding items. Is that correct?

**Comment by aarowman** [comment]
It's about the bug that it has an error or has undesired behavior. I can confirm the scenario of adding children - if the scroll may still fail, e.g.

1. start with 10 children
2. scroll to last page, 
3. then update children to 15 
4. see if the NEXT arrow works

**Comment by sag1v** [comment]
@aarowman Thanks! 
Fixed, available as of version `0.6.1` 🙂
